query-engine/js-connectors/src/conversion.rs

```diff
@@ -0,0 +1,59 @@
+use napi::bindgen_prelude::{FromNapiValue, ToNapiValue};
+use quaint::ast::Value as QuaintValue;
+use serde::Serialize;
+use serde_json::value::Value as JsonValue;
+
+#[derive(Debug, Serialize)]
+#[serde(untagged)]
+pub enum JSArg {
+    RawString(String),
+    Value(serde_json::Value),
+}
+
+impl From<JsonValue> for JSArg {
+    fn from(v: JsonValue) -> Self {
+        JSArg::Value(v)
+    }
+}
+
+// FromNapiValue is the napi equivalent to serde::Deserialize.
```

Comment:
Do we need to implement this trait? I think, unless we are using `JSArg` in `#[napi(object]` we should be fine with just `ToNapiValue`

Reply:
Unfortunately we do. Removing it would result in:

```
error[E0277]: the trait bound `Vec<JSArg>: napi::bindgen_prelude::FromNapiValue` is not satisfied
   --> query-engine/js-connectors/src/proxy.rs:152:1
    |
152 | #[napi(object)]
    | ^^^^^^^^^^^^^^^ the trait `napi::bindgen_prelude::FromNapiValue` is not implemented for `Vec<JSArg>`
    |
    = help: the trait `napi::bindgen_prelude::FromNapiValue` is implemented for `Vec<T>`
note: required by a bound in `napi::bindgen_runtime::js_values::object::<impl JsObject>::get`
   --> /Users/jkomyno/.cargo/registry/src/index.crates.io-6f17d22bba15001f/napi-2.13.2/src/bindgen_runtime/js_values/object.rs:24:32
    |
24  |   pub fn get<K: AsRef<str>, V: FromNapiValue>(&self, field: K) -> Result<Option<V>> {
    |                                ^^^^^^^^^^^^^ required by this bound in `napi::bindgen_runtime::js_values::object::<impl JsObject>::get`
    = note: this error originates in the attribute macro `napi` (in Nightly builds, run with -Z macro-backtrace for more info)
```